pkg/zonemgr/zone_manager.go

```diff
@@ -41,7 +39,6 @@ func (zoneMgr *ZoneManager) prepare() {
 	nameServerIP := os.Getenv(envVarNameServerIP)
 
 	zoneMgr.zoneFileCache = NewZoneFileCache(nameServerIP, domain, zoneFileNamePrefix)
-	zoneMgr.zoneFile = NewZoneFile(zoneFileNamePrefix + zoneMgr.zoneFileCache.Domain)
```

Comment:
The ZoneFileManager feels pretty redundant now it manages only one component. Also the name `ZoneFileCache` is not accurate any more.
I like more the previous design. It was clear we have a file and a cache and the manager is managing both.

Reply:
In this case we are back with the SOA serial issue - either override it from the ZoneManager if necessary (which I prefer) or fetch from inside ZoneFileCache (which breaks the design IMO)
Unless you have another idea

Reply:
please see the two ideas here @AlonaKaplan @dteplits 
https://github.com/kubevirt/kubesecondarydns/pull/45#discussion_r1051636728

Reply:
I don't fully understand the SOA serial issue. Why cannot the zoneManager read it via the zoneFile and pass it to the cache?